en/mulesoft/modules/ROOT/pages/ms_composer_flows.adoc

```diff
@@ -334,6 +334,103 @@ When there are null values in custom expressions and are:
+. In the *Action* dropdown, click *Create Variable*.
+. In the *Variable* section, click *Add Variable*. 
+. In the *Visual Editor*, in the *Label* field, enter a unique name for the variable. Note that the name entered is only unique to this step. Values in the *Label* field can be duplicated across different Variable steps. 
+. In the *Data Type* drop-down, select the data type you want to use for the variable. The following xref:ms_composer_custom_expression_editor.adoc#data-types[data types] are supported: `String`, `Number`, `Boolean`, `Date`, `DateTime`.
+. Optionally, to add additional variables in the *Visual Editor*, click *Add Variable*. You can create multiple variables in each Variable step. 
+. Click *Apply*. +
+The *Create Variables* window appears, displaying one or more new variables added to the flow. 
+. Optionally, to make changes to the new variables, click *Manage Variables* to open the Visual Editor or *Edit Schema* to open the JSON Schema. 
+. Complete the initial value for each variable. The default value is defined under each variable. If the field empty, the value is set to the default value for that data type. You can enter a value manually, select any existing data pill (including variables) from your flow, or you can select the xref:ms_composer_custom_expression_editor.adoc#add-custom-expression-to-an-action[*Custom Expression Editor*] to create a custom expression. 
+
+=== Create a Variable Using the JSON Schema
+
+To create one or more new variables using the *JSON Schema* window: 
+
+. <<access-the-variable-step,Access the *Variable* step.>> 
+. In the *Action* dropdown, click *Create Variable*.
+. In the *Variable* section, click *Create Schema*. 
+. In the *JSON Schema* window, enter the properties for the variable you want to create. Note that: 
+.. Each variable is defined within a `Properties`  object with a `Variable Unique ID`.  The `Variable Unique ID` allows the variable to be tracked within your flow. The `Variable Unique ID` can only contain alphanumeric characters, hyphens, and underscores. When using the Visual Editor, this ID is automatically generated for you.
+Each variable must specify the `Name` and data type. To set the properties for the variable definition, complete the following:
+... `Variable Label (Title)`:  The display label for the variable. A value is required. 
+... `Type` : The data type of the variable. The following xref:ms_composer_custom_expression_editor.adoc#data-types[data types] are supported: `String`, `Number`, and `Boolean`. To define a `Date` or `DateTime` variable, the data type should be set to `String` and the `Format` property should be defined. A value is required. 
+... `Format`:  The qualifier for `String` properties that can specify additional data types not natively supported in JSON. For variables, this must be set when defining `Date` or `DateTime` variables as `date` or `date-time` respectively. A value in this field is optional. 
```

Comment:
```suggestion
... `Format`:  The qualifier for `String` properties that can specify additional data types not natively supported in JSON. For variables, this value must be set when defining `Date` or `DateTime` variables as `date` or `date-time` respectively. (Optional)
```

Reply:
I'm thinking, by putting just (Optional) or (Required), it will be easier for the reader to scan.

Reply:
Done!